src/der.rs

```diff
@@ -311,10 +311,14 @@ pub(crate) fn nested_of_mut<'a>(
     outer_tag: Tag,
     inner_tag: Tag,
     error: Error,
+    allowed_empty: AllowedToBeEmpty,
```

Comment:
Name this actively, like `allow_empty`? (Also the enum, except:)

Are you sure the custom enum is worth it here? Not sure this is a context where using a `bool` would incur substantial risk.